devito/ir/iet/utils.py

```diff
@@ -110,7 +110,9 @@ def derive_parameters(nodes, drop_locals=False):
     parameters = tuple(s for s in symbols if s.name not in defines)
 
     # Drop globally-visible objects
-    parameters = [p for p in parameters if not isinstance(p, (Literal, Macro))]
+    parameters = [
+        p for p in parameters if not isinstance(p, (DefaultDimension, Literal, Macro))
```

Comment:
are we sure this is necessary?

Reply:
Another option would be overriding `free_symbols` in Array to not return `DefaultDimension`s

Reply:
I think I'd prefer that?

after all, we do not generate the DefaultDimension right?